presto-main/src/main/java/com/facebook/presto/security/FileBasedSystemAccessControl.java

```diff
@@ -103,6 +111,29 @@ public SystemAccessControl create(Map<String, String> config)
     @Override
     public void checkCanSetUser(Principal principal, String userName)
     {
+        if (principal == null) {
+            return;
+        }
+
+        String principalName = principal.getName();
+
+        if (usernameExtractionRules.isEmpty()) {
+            return;
+        }
+
+        for (Pattern pattern : usernameExtractionRules) {
+            Matcher matcher = pattern.matcher(principalName);
+            while (matcher.find()) {
```

Comment:
Use `pattern.matches()` here.
We do not want too search for multiple occurences of pattern in principal, but rather check if principal matches regex given in configuration.

Or, do you see use-case when using `find()` would be useful?